### DISCUSSION TITLE
Mutual Information Matrix for SU(2) Symmetry

### DISCUSSION THREAD

**Discussion by jtcantin** [block-hczhai/block2-preview#99] [Q&A] [answered]
Hello Huanchen,

I would like to use the Fiedler vector to reorder orbitals based on the mutual information matrix for spin-adapted calculations (SU(2)).

However, I am getting errors when I try to run `get_orbital_interaction_matrix` on a ket from a small bond dimension SU(2) calculation. I can do so successfully for Sz symmetry, but the reordering I get from this seems to cause problems for the SU(2) calculation (poorer convergence behaviour, for example).

Is there a way to get the mutual information matrix based on the DMRG sites used in the SU(2) calculation (which I suppose would be a "CSF interaction matrix" as opposed to an "orbital interaction matrix")?

Thanks,
Josh

**Comment by hczhai** [comment, marked as answer]
The implementation of mutual information is based on the literature: *J. Chem. Theory Comput.* **9**, 2959-2973 (2013).

There is no mutual information matrix implementation in the SU(2) mode simply because I am not aware of any published research work describing how to do it.

Of course you can translate the MPS from SU2 to SZ (https://block2.readthedocs.io/en/latest/tutorial/qc-hamiltonians.html#Change-from-SU2-MPS-to-SZ-MPS) and get the mutual information. And then you can use the mutual information for reordering in a subsequent SU2 calculation.

**Reply by jtcantin** [reply]
Thank you!
